app/views/challenge/shared/_challenge_form_delete.html.erb

```diff
@@ -1,12 +1,15 @@
-<ul class="navigation actions" role="menu">
-  <% if @collection.challenge %>
-    <li><%= link_to ts("Delete Challenge"),
-                    if @collection.prompt_meme?
-                      collection_prompt_meme_path(@collection)
-                    else
-                      collection_gift_exchange_path(@collection)
-                    end,
-                    data: {confirm: ts('Are you certain you want to delete the challenge from this collection? All signups, assignments, and settings will be lost. (Works will remain in the collection.)')},
-                    :method => :delete %></li>
-  <% end %>
-</ul>
+<% 
+  # This is a list element that should always be rendered within an ul with classes "navigation actions" and role "menu".
```

Comment:
The `menu` role is actually a relic that should be removed -- the way we use it is not correct at all. Could you remove them from the `ul` elements and update this note?

And, actually, it would probably be better to move the `li` tags out of this file in case we ever want to use the link outside a `ul`. That's what we do with [the subscribe button](https://github.com/otwcode/otwarchive/blob/master/app/views/subscriptions/_form.html.erb#L1), for example.

Reply:
As per the commit messages, I did the following:

- remove role attributes from ul
- remove li from shared form and add in the views them where necessary
- add final newlines
- change confirmation text for confirm_delete page, Javascript prompt
- fix indent on steps in gift exchange feature
- change confirmation text in features for all challenge types
- add spaces inside braces on forms for _delete and _confirm_delete
- switch to double quotes in routes.rb

I hope I got them all!